src/haddock/modules/analysis/caprieval/__init__.py

```diff
@@ -44,51 +73,79 @@ def _run(self):
-            aln_method=self.params["alignment_method"],
-            path=Path("."),
-            lovoalign_exec=self.params["lovoalign_exec"],
-            )
-
-        if self.params["fnat"]:
-            self.log("Calculating FNAT")
-            fnat_cutoff = self.params["fnat_cutoff"]
-            self.log(f" cutoff: {fnat_cutoff}A")
-            capri.fnat(cutoff=fnat_cutoff)
-
-        if self.params["irmsd"]:
-            self.log("Calculating I-RMSD")
-            irmsd_cutoff = self.params["irmsd_cutoff"]
-            self.log(f" cutoff: {irmsd_cutoff}A")
-            capri.irmsd(cutoff=irmsd_cutoff)
-
-        if self.params["lrmsd"]:
-            self.log("Calculating L-RMSD")
-            capri.lrmsd()
-
-        if self.params["ilrmsd"]:
-            self.log("Calculating I-L-RMSD")
-            ilrmsd_cutoff = self.params["irmsd_cutoff"]
-            self.log(f" cutoff: {ilrmsd_cutoff}A")
-
-            capri.ilrmsd(
-                cutoff=ilrmsd_cutoff,
+        # Parallelisation : optimal dispatching of models
+        nmodels = len(models)
+        ncores = parse_ncores(n=self.params['ncores'], njobs=nmodels)
+        base_models = nmodels // ncores
+        modulo = nmodels % ncores
+        chain_of_idx = [0]
+        for core in range(ncores):
+            if core < modulo:
+                chain_of_idx.append(chain_of_idx[core] + base_models + 1)
+            else:
+                chain_of_idx.append(chain_of_idx[core] + base_models)
+        # starting jobs
+        capri_jobs = []
+        cluster_info = []
+        self.log(f"running Capri Jobs with {ncores} cores")
+        for core in range(ncores):
+            # init Capri
+            capri_obj = CAPRI(
+                reference,
+                models[chain_of_idx[core]:chain_of_idx[core + 1]],
+                receptor_chain=self.params["receptor_chain"],
+                ligand_chain=self.params["ligand_chain"],
+                aln_method=self.params["alignment_method"],
+                path=Path("."),
+                lovoalign_exec=self.params["lovoalign_exec"],
+                core=core,
+                core_model_idx=chain_of_idx[core]
                 )
-
-        if self.params["dockq"]:
-            self.log("Calculating DockQ metric")
-            capri.dockq()
-
-        self.log(" Saving output")
-        capri.output(
-            self.params["clt_threshold"],
-            sortby_key=self.params["sortby"],
-            sort_ascending=self.params["sort_ascending"],
+            # Name job
+            job_f = Path("capri_ss_" + str(core) + ".tsv")
+            # Get cluster info
+            cluster_info.append(any(m.clt_id for m in capri_obj.model_list))
+            # init CapriJob
+            job = CapriJob(
+                job_f,
+                self.params,
+                capri_obj
+                )
+            capri_jobs.append(job)
+        # Running parallel Capri Jobs
+        capri_engine = Scheduler(capri_jobs, ncores=ncores)
```

Comment:
@rvhonorato do you think here we could use the `get_engine` function to run `capri` in `hpc` if the user uses that mode? Or is it too much?

@mgiulini you would implement it like this:

```python
from haddock.modules import get_engine
self.params["ncores"] = ncores
Engine = get_engine(self.params["mode"], self.params)
engine = Engine(capri_jobs)
engine.run()
```

Reply:
I think the overhead of sending it to the queue and then waiting for it to be picked up by the queue manager will be much longer than the execution.

But its a good point because if we are running it via HPC, would it be faster to run in sequence or to indeed send it to the queue? Something we should test

Reply:
@rvhonorato agree with you! I am going to check the difference anyway